todo_app/app.py

```diff
@@ -1,40 +1,249 @@
-from flask import Flask, render_template, redirect, url_for, request
+import os, requests
+from flask import Flask, redirect, url_for, session, request
+from todo_app.managers.DbConnectionManager import *
+from todo_app.managers.UserManager import *
+from todo_app.models.view import View
+from todo_app.models.card import Card
+from todo_app.models.user import User
+# Login
+from flask_login import UserMixin, LoginManager, login_required, login_user, logout_user
+from oauthlib.oauth2 import WebApplicationClient
 
-from todo_app.ViewModel import ViewModel
-from todo_app.data.Trello import trello
-from todo_app.flask_config import Config
+
+class ReverseProxied(object):
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        scheme = environ.get('HTTP_X_FORWARDED_PROTO')
+        if scheme:
+            environ['wsgi.url_scheme'] = scheme
+        return self.app(environ, start_response)
+
+
+class UserToLogin(UserMixin):
+    def __init__(self, id):
+        self.id = id
```

Comment:
You shouldn't have two User classes. Just have your full `User` class extend UserMixin. When you first log them in during the callback, you can create the entity in the DB, convert it to a User and log them in.

In your load_user function, read them out of the database.

That way you can super easily use `current_user` to check the current user's role etc instead of the collection query logic you had to put in a few places.

Only change is you may need your User class to have a function called id that returns self._id.